shared/README.md

```diff
@@ -0,0 +1,110 @@
+
+## Vehicle Events
+
+This is the list of `vehicle_state` and `event_type` pairings that constitute the valid transitions of the vehicle state machine.
+
+Note that to handle out-of-order events, the validity of the prior-state is not enforced at the time of ingest via Provider or Agency.  Events received out-of-order may result in transient incorrect vehicle states.
+
+| `vehicle_state` | `event_type` | Valid prior `vehicle_state` values | Description |
+| --- | --- | --- | --- |
+| `available` | `battery_charged` | `unavailable` | The vehicle became available because its battery is now charged. |
+| `available` | `on_hours` | `unavailable` | The vehicle has entered operating hours (per the regulator or per the provider) |
+| `available` | `provider_drop_off` | `removed` | The vehicle was placed in the PROW by the provider |
+| `available` | `agency_drop_off` | `removed` | The vehicle was placed in the PROW by a city or county |
+| `available` | `maintenance` | `unavailable` | The vehicle was previously in need of maintenance |
+| `available` | `trip_end` | `trip` | A trip has ended, and the vehicle is again available for rent |
+| `available` | `reservation_cancel` | `reserved` | A reservation was canceled and the vehicle returned to service |
+| `available` | `trip_cancel` | `trip` | A trip was initiated, then canceled prior to moving any distance |
+| `available` | `unspecified` | `unavailable`, `unknown`, `removed`, `reserved`, `elsewhere` | The vehicle became available, but the provider cannot definitively (yet) specify the reason.  Generally, regulator Service-Level Agreements will limit the amount of time a vehicle's last event type may be `unspecified`. |
+| `reserved` | `reservation_start` | `available` | The vehicle was reserved for use by a customer |
+| `trip` | `trip_start` | `available`, `reserved` | A customer initiated a trip with this vehicle |
+| `trip` | `trip_enter_jurisdiction` | `elsewhere` | A vehicle on a trip entered the jurisdiction |
+| `elsewhere` | `trip_leave_jurisdiction` | `trip` | A vehicle on a trip left the jurisdiction |
+| `unavailable` | `low_battery` | `available` | The vehicle's battery is below some rentability threshold |
+| `unavailable` | `maintenance` | `available` | The vehicle requires some non-charge-related maintenance |
+| `unavailable` | `off_hours` | `available` | The vehicle has exited operating hours (per the regulator or per the Provider) |
+| `unavailable` | `unspecified` | `available` | The vehicle became unavailable, but he Provider cannot definitively (yet) specify the reason. |
+| `removed` | `rebalance_pick_up` | `available`, `unavailable`, `elsewhere` | The provider picked up the vehicle for rebalancing purposes |
+| `removed` | `maintenance_pick_up` | `available`, `unavailable`, `elsewhere` | The provider picked up the vehicle to service it |
+| `removed` | `agency_pick_up` | any | An agency picked up the vehicle for some reason, e.g. illegal placement |
```

Comment:
Why would pickups be represented for `elsewhere`?  That seems to indicate a need to report outside of jurisdiction.

Reply:
Do you want me to change this to "any except `elsewhere`"?